packages/backend/src/models/CatalogModel/CatalogModel.ts

```diff
@@ -852,4 +854,16 @@ export class CatalogModel {
             metricTreeEdgesMigrateIn,
         );
     }
+
+    async getAllMetricFields(
+        projectUuid: string,
+        userAttributes: UserAttributeValueMap,
+    ) {
+        return this.database(CatalogTableName)
+            .where('project_uuid', projectUuid)
+            .andWhere('type', CatalogType.Field)
+            .andWhere('field_type', FieldType.METRIC)
+            .andWhere(userAttributesFiltering(userAttributes))
```

Comment:
maybe you want tablesConfigurationFiltering as well...?

Reply:
or just use the `search method...`?

Reply:
can I search without search query? i.e. all metrics?